src/Donations/Endpoints/DonationUpdateAttributes/DonationUpdateAttribute.php

```diff
@@ -0,0 +1,28 @@
+<?php
+
+namespace Give\Donations\Endpoints\DonationUpdateAttributes;
+
+use Give\Donations\Models\Donation;
+
+abstract class DonationUpdateAttribute
+{
+    /**
+     * @return string
+     */
+    abstract public static function getId(): string;
+
+    /**
+     * @return array
+     */
+    abstract public static function getDefinition(): array;
+
+    /**
+     * @param          $value
+     * @param Donation $donation
+     *
+     * @return void
+     */
+    public static function update($value, Donation $donation)
+    {
+    }
```

Comment:
@pauloiankoski Why not make this an abstract function?

Reply:
In case we have attributes that are read-only (eg: IdAttribute) we would need to include that method as empty as an abstract method requires implementation.

Reply:
Resolved a73180607c598027e3d9f157968f03873e24668f